### PR TITLE
Log entitlement agent attach timing

### PR DESCRIPTION
**Pull request by jdconrad**
- Add an INFO log line on every successful entitlement agent attach, breaking out total / `VirtualMachine.attach` / `loadAgent`+`detach` timings, so future occurrences of the slow-attach pattern in elastic/elasticsearch-team#4184 are diagnosable.
- Include elapsed milliseconds in the `IllegalStateException` thrown when the JVM attach handshake fails.
- Does not change the JVM attach timeout.

### CODE REVIEW

File: libs/entitlement/src/main/java/org/elasticsearch/entitlement/bootstrap/EntitlementBootstrap.java [hunk 1]
```diff
@@ -127,17 +136,35 @@ private static Path getUserHome() {
...
+        long startNanos = System.nanoTime();
         try {
             VirtualMachine vm = VirtualMachine.attach(Long.toString(ProcessHandle.current().pid()));
+            long attachedNanos = System.nanoTime();
             try {
                 vm.loadAgent(agentPath, entitlementInitializationClassName);
             } finally {
                 vm.detach();
             }
+            long doneNanos = System.nanoTime();
```
Comment (prdoyle) [inline libs/entitlement/src/main/java/org/elasticsearch/entitlement/bootstrap/EntitlementBootstrap.java]: If we're computing milliseconds, `currentTimeMillis()` seems more straightforward than asking for nanos and then dividing by a million.

**Review by prdoyle — DISMISSED** [review]
10 seconds to start up Elasticsearch does seem to be a tad tight.

Optional comment below.

**Comment by mark-vieira** [comment]
This is concerning. This isn't 10 seconds to "start elasticsearch" this is 10 seconds just to attach the entitlement agent. That implies the recent refactoring has made this substantially more expensive.

**Comment by jdconrad** [comment]
@mark-vieira It's only happened twice in a 2 week period when I was looking through logs. I debated extending it at all. Or if we should just have a new logging message. My guess it happens when a node is already having issues from the beginning. Would appreciate your thoughts on if we should just add the logging message for timing.

File: libs/entitlement/src/main/java/org/elasticsearch/entitlement/bootstrap/EntitlementBootstrap.java (hunk 1 above)
Comment (jdconrad) [reply libs/entitlement/src/main/java/org/elasticsearch/entitlement/bootstrap/EntitlementBootstrap.java]: I'll update this. Thanks!

**Comment by mark-vieira** [comment]
> @mark-vieira It's only happened twice in a 2 week period when I was looking through logs.

What logs? Is this happening in production or in tests?

**Comment by mark-vieira** [comment]
> Would appreciate your thoughts on if we should just add the logging message for timing.

I'm leaning towards starting with logging so we have a baseline of how long this takes typically before we make assumptions about these being one-off pathological cases. If the average is 1 second and a couple hit the 10 second timeout then that's definitely something up with the nodes. If it's typically taking 5+ seconds and we occassionally hit the timeout that's very different.

**Comment by prdoyle** [comment]
> This isn't 10 seconds to "start elasticsearch" this is 10 seconds just to attach the entitlement agent.

Oh good catch! I stand corrected.

**Comment by jdconrad** [comment]
Okay, I'll just log how long it takes the agent to attach for now. This is happening in production logs for serverless, but very rarely. See: https://github.com/elastic/elasticsearch-team/issues/4184

**Review by mark-vieira — APPROVED** [review]
